### ISSUE TITLE
Handling Errors from OpenAI API

### ISSUE THREAD

**Issue by avirlrma** [weaviate/weaviate#4080]
### How to reproduce this bug?

I've tested with `qna-openai` module, possibly this would be occurring in other modules as well.

create a schema with default model: `text-ada-001`.  

Query the collection:

```
curl --location 'http://localhost:8080/v1/graphql' \
--header 'Content-Type: application/json' \
--data '{ "query": "{ Get { Article( ask: { question: \"Who is Stanley Kubrick?\", properties: [\"title\"] }, limit: 1 ) { title _additional { answer { hasAnswer property result startPosition endPosition } } } } }" }'

``` 

### What is the expected behavior?

Any errors from openAI API should be gracefully handled.

### What is the actual behavior?

```
{
    "data": {
        "Get": {
            "Article": null
        }
    },
    "errors": [
        {
            "locations": [
                {
                    "column": 9,
                    "line": 1
                }
            ],
            "message": "explorer: get class: extend: extend answer: unmarshal response body: json: invalid number literal, trying to unmarshal \"\\\"model_not_found\\\"\" into Number",
            "path": [
                "Get",
                "Article"
            ]
        }
    ]
}

```

### Server Version

1.23.3

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by avirlrma** [comment]
Corresponding loc from openai-python-client: https://github.com/openai/openai-python/blob/main/src/openai/_exceptions.py#L28 indicating `code` is a string.

**Comment by antas-marcin** [comment]
This issue is fixed in `v1.23.5` release ([PR](https://github.com/weaviate/weaviate/pull/4068) with fix)
